presto-hive/src/test/java/com/facebook/presto/hive/TestAriaHiveDistributedQueries.java

```diff
@@ -435,4 +446,15 @@ public void testJoins()
                 "JOIN partsupp p\n" +
                 "    ON p.suppkey = l.suppkey AND p.partkey = l.partkey");
     }
+
+    @Test
+    public void testFilteredErrors()
+    {
+        // Two filters which produce errors so that when one has an error the other is false.
+        assertQuery(ariaSession(),
+                    "select count (*) from lineitem where \n" +
+                    "if (linenumber = 2, false, suppkey / (linenumber - 3) >= 0) \n" +
+                    "and if (linenumber = 3, false, partkey / (linenumber - 2) >= 0)",
+                    noAriaSession(), Optional.of("select count (*) from lineitem where linenumber > 3"));
```

Comment:
The "expected" query is very simple and is fully supported by H2. Hence, no need to use the new assertQuery method. The following should work just fine.

```
       assertQuery(ariaSession(),
                    "select count (*) from lineitem where \n" +
                    "if (linenumber = 2, false, suppkey / (linenumber - 3) >= 0) \n" +
                    "and if (linenumber = 3, false, partkey / (linenumber - 2) >= 0)",
                    "select count (*) from lineitem where linenumber > 3");
```

I'm having doubts about testing aria by running a query with and without aria. First, I don't think we can trust non-aria path if the query can't be validated against a third-party, e.g. H2. Second, if we do compare two flavors of Presto, then I think we should run the same query. Here, we are running two different queries, presumably, because filter error handling is different between the two systems.